DEV_GUIDE.md

```diff
@@ -26,6 +27,22 @@ This document gives a detailed breakdown of the various build processes and opti
   * [Rendering documentation](#rendering-documentation)
 <!-- TOC -->
 
+## Project structure
+
+The module structure is as follows:
+
+* `kroxylicious-bom` the Bill Of Materials used by developers to conveniently declare a consistent set of Kroxylicious dependencies.
+* `krpc-code-gen` a code generator used to build the API module.
+* `kroxylicious-api` the APIs needed to write a filter.
+* `kroxylicious-unit-test` test utilities for filter authors.
+* `kroxylicious` the proxy runtime platform for hosting filters, including the networking code etc, but lacking a `main()` method
```

Comment:
I think we should rename this `kroxylicious-core` so it fits with the `kroxylicious-*` pattern.

Reply:
Given you just described it as `the proxy runtime` I think `kroxylicious-runtime` might be a better name.

Reply:
Captured as #637